eras/conway/impl/src/Cardano/Ledger/Conway/TxBody.hs

```diff
@@ -1,145 +1,532 @@
+lensTxBodyRaw getter setter =
+  lens
+    (\(TxBodyConstr (Memo txBodyRaw _)) -> getter txBodyRaw)
+    (\(TxBodyConstr (Memo txBodyRaw _)) val -> mkConwayTxBodyFromRaw $ setter txBodyRaw val)
+{-# INLINEABLE lensTxBodyRaw #-}
 
 instance CC.Crypto c => EraTxBody (ConwayEra c) where
   {-# SPECIALIZE instance EraTxBody (ConwayEra CC.StandardCrypto) #-}
 
-  type TxBody (ConwayEra c) = BabbageTxBody (ConwayEra c)
+  type TxBody (ConwayEra c) = ConwayTxBody (ConwayEra c)
 
-  mkBasicTxBody = mkBabbageTxBody
+  mkBasicTxBody = mkConwayTxBodyFromRaw initialTxBodyRaw
 
-  inputsTxBodyL = inputsBabbageTxBodyL
+  inputsTxBodyL = lensTxBodyRaw ctbrSpendInputs (\txb x -> txb {ctbrSpendInputs = x})
   {-# INLINE inputsTxBodyL #-}
 
-  outputsTxBodyL = outputsBabbageTxBodyL
+  outputsTxBodyL = lensTxBodyRaw (fmap sizedValue . ctbrOutputs) (\txb x -> txb {ctbrOutputs = mkSized (eraProtVerLow @(ConwayEra c)) <$> x})
   {-# INLINE outputsTxBodyL #-}
 
-  feeTxBodyL = feeBabbageTxBodyL
+  feeTxBodyL = lensTxBodyRaw ctbrTxfee (\txb x -> txb {ctbrTxfee = x})
   {-# INLINE feeTxBodyL #-}
 
-  auxDataHashTxBodyL = auxDataHashBabbageTxBodyL
+  auxDataHashTxBodyL = lensTxBodyRaw ctbrAdHash (\txb x -> txb {ctbrAdHash = x})
   {-# INLINE auxDataHashTxBodyL #-}
 
-  allInputsTxBodyF = allInputsBabbageTxBodyF
+  allInputsTxBodyF =
+    to $ \txBody ->
+      Set.unions
+        [ txBody ^. inputsTxBodyL,
+          txBody ^. collateralInputsTxBodyL,
+          txBody ^. referenceInputsTxBodyL
+        ]
   {-# INLINE allInputsTxBodyF #-}
 
 instance CC.Crypto c => ShelleyEraTxBody (ConwayEra c) where
   {-# SPECIALIZE instance ShelleyEraTxBody (ConwayEra CC.StandardCrypto) #-}
 
-  wdrlsTxBodyL = wdrlsBabbbageTxBodyL
+  wdrlsTxBodyL = lensTxBodyRaw ctbrWdrls (\txb x -> txb {ctbrWdrls = x})
   {-# INLINE wdrlsTxBodyL #-}
 
   ttlTxBodyL = notSupportedInThisEraL
   {-# INLINE ttlTxBodyL #-}
 
-  updateTxBodyL = updateBabbageTxBodyL
+  updateTxBodyL = notSupportedInThisEraL
   {-# INLINE updateTxBodyL #-}
 
-  certsTxBodyL = certsBabbageTxBodyL
+  certsTxBodyL = notSupportedInThisEraL
   {-# INLINE certsTxBodyL #-}
 
-instance CC.Crypto c => AllegraEraTxBody (ConwayEra c) where
-  {-# SPECIALIZE instance AllegraEraTxBody (ConwayEra CC.StandardCrypto) #-}
-
-  vldtTxBodyL = vldtBabbageTxBodyL
-  {-# INLINE vldtTxBodyL #-}
-
 instance CC.Crypto c => MaryEraTxBody (ConwayEra c) where
   {-# SPECIALIZE instance MaryEraTxBody (ConwayEra CC.StandardCrypto) #-}
 
-  mintTxBodyL = mintBabbageTxBodyL
+  mintTxBodyL = lensTxBodyRaw ctbrMint (\txb x -> txb {ctbrMint = x})
   {-# INLINE mintTxBodyL #-}
 
-  mintValueTxBodyF = mintValueBabbageTxBodyF
+  mintValueTxBodyF = mintTxBodyL . to (MaryValue 0)
 
-  mintedTxBodyF = mintedBabbageTxBodyF
+  mintedTxBodyF = to (\(TxBodyConstr (Memo txBodyRaw _)) -> Set.map policyID (policies (ctbrMint txBodyRaw)))
   {-# INLINE mintedTxBodyF #-}
 
+instance CC.Crypto c => AllegraEraTxBody (ConwayEra c) where
```

Comment:
Not terribly important, but `Allegra` came before `Mary`, so for consistency sake it might be better to move it back up.